spec/rubocop/cop/style/redundant_map_compact_spec.rb

```diff
@@ -0,0 +1,389 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::RedundantMapCompact, :config do
+  context 'With multiline block' do
+    it 'registers an offense and corrects to `select` with `if` condition' do
+      expect_offense(<<~RUBY)
+        foo.map do |item|
+            ^^^^^^^^^^^^^ Use `select { ... }` instead of `map { ... }.compact`.
```

Comment:
The message looks simple as follows.

```suggestion
            ^^^^^^^^^^^^^ Use `select` instead`.
```

Reply:
I'm fine with the more descriptive message, as some people might forget to remove the `compact` otherwise. "Replace X with Y." is another wording that I like.